vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextFieldBase.java

```diff
@@ -45,7 +45,7 @@
  * @param <TValue>
  *            Type of the value of the extending component
  */
-public abstract class InternalFieldBase<TComponent extends InternalFieldBase<TComponent, TValue>, TValue>
+public abstract class TextFieldBase<TComponent extends TextFieldBase<TComponent, TValue>, TValue>
```

Comment:
`TextFieldBase` could now be made package-private if we don't intend to expose it.

Reply:
Well that might be a problem if you want to take a collection of fields, and do something generic with them, like generating placeholders. You would not be able to cast to `TextFieldBase` then.

Reply:
Well, turns out this class isn't really internal considering the use-case above.

Reply:
Yes, but it's probably fine as is. This is definitely public API, because developers are going to cast to `TextFieldBase`, and the methods in it also need to be stable. However making the constructor package-private is good, that gives us some guarantee that the implementation isn't used in custom classes.